### PR TITLE
Fix #113: Add support for implicit login operation

### LINKED ISSUE #113: Add support for implicit login operation

### PR DESCRIPTION
**Pull request by petrdvorak**
Example implementation works with `aisp` and `pisp` scopes...

### CODE REVIEW

File: powerauth-data-adapter/src/main/java/io/getlime/security/powerauth/app/dataadapter/impl/service/DataAdapterService.java [hunk 1]
```diff
@@ -190,6 +190,46 @@ public void formDataChangedNotification(String userId, String organizationId, Fo
...
     }
 
+    @Override
+    public CreateImplicitLoginOperationResponse createImplicitLoginOperation(String clientId, String[] scopes) throws DataAdapterRemoteException {
+        try {
+            final TppAppDetailResponse appDetail = tppEngineService.fetchAppDetail(clientId);
+
+            // Make sure there is only one item in scopes
+            if (scopes == null || scopes.length != 1) {
+                return null;
```
Comment (romanstrobl) [inline powerauth-data-adapter/src/main/java/io/getlime/security/powerauth/app/dataadapter/impl/service/DataAdapterService.java]: Returning `null` seems a bit weird. Why not throw the exception instead? Furthermore, the error reason should probably be logged.

File: powerauth-data-adapter/src/main/java/io/getlime/security/powerauth/app/dataadapter/impl/service/DataAdapterService.java [hunk 2]
```diff
@@ -190,6 +190,46 @@ public void formDataChangedNotification(String userId, String organizationId, Fo
...
+            }
+            // Make sure the scope is from known enum
+            final String scope = scopes[0].toLowerCase();
+            if (!"aisp".equals(scope) && !"pisp".equals(scope)) {
+                return null;
```
Comment (romanstrobl) [inline powerauth-data-adapter/src/main/java/io/getlime/security/powerauth/app/dataadapter/impl/service/DataAdapterService.java]: dtto

File: powerauth-data-adapter/src/main/java/io/getlime/security/powerauth/app/dataadapter/impl/service/DataAdapterService.java [hunk 3]
```diff
@@ -190,6 +190,46 @@ public void formDataChangedNotification(String userId, String organizationId, Fo
...
+            formData.addSummary("login.summary");
+
+            // Create an implicit operation
+            CreateImplicitLoginOperationResponse result = new CreateImplicitLoginOperationResponse();
+            result.setName("login_sca");
+            result.setFormData(formData);
+            result.setApplicationContext(appContext);
+
+            return result;
+        } catch (InvalidAppException e) {
```
Comment (romanstrobl) [inline powerauth-data-adapter/src/main/java/io/getlime/security/powerauth/app/dataadapter/impl/service/DataAdapterService.java]: Where does this exception come from? Perhaps I am missing some code changes in `powerauth-webflow`.

Comment (petrdvorak) [reply powerauth-data-adapter/src/main/java/io/getlime/security/powerauth/app/dataadapter/impl/service/DataAdapterService.java]: ... `features/psd2` branch...

**Pull request merged**